concrete-core/src/backends/core/implementation/engines/ggsw_ciphertext_conversion.rs

```diff
@@ -0,0 +1,200 @@
+    /// use concrete_commons::parameters::{
+    ///     DecompositionBaseLog, DecompositionLevelCount, GlweDimension, LweDimension, PolynomialSize,
+    /// };
+    /// use concrete_core::prelude::*;
+    /// # use std::error::Error;
+    ///
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// // DISCLAIMER: the parameters used here are only for test purpose, and are not secure.
+    /// let glwe_dimension = GlweDimension(2);
+    /// let polynomial_size = PolynomialSize(256);
+    /// let level = DecompositionLevelCount(1);
+    /// let base_log = DecompositionBaseLog(4);
+    /// // Here a hard-set encoding is applied (shift by 20 bits)
+    /// let input = 3_u32 << 20;
+    /// let noise = Variance(2_f64.powf(-25.));
+    ///
+    /// let mut engine = CoreEngine::new()?;
+    /// let key: GlweSecretKey32 = engine.create_glwe_secret_key(glwe_dimension, polynomial_size)?;
+    /// let plaintext = engine.create_plaintext(&input)?;
+    ///
+    /// let ciphertext = engine.encrypt_ggsw_ciphertext(&key, &plaintext, noise, level, base_log)?;
+    ///
+    /// let fourier_ciphertext: GgswCiphertextComplex64 =
+    ///     engine.convert_ggsw_ciphertext(&ciphertext)?;
+    /// #
+    /// assert_eq!(fourier_ciphertext.glwe_dimension(), glwe_dimension);
+    /// assert_eq!(fourier_ciphertext.polynomial_size(), polynomial_size);
+    /// assert_eq!(fourier_ciphertext.decomposition_base_log(), base_log);
+    /// assert_eq!(fourier_ciphertext.decomposition_level_count(), level);
+    ///
+    /// engine.destroy(key)?;
+    /// engine.destroy(plaintext)?;
+    /// engine.destroy(ciphertext)?;
+    /// engine.destroy(fourier_ciphertext)?;
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    fn convert_ggsw_ciphertext(
+        &mut self,
+        input: &GgswCiphertext32,
+    ) -> Result<GgswCiphertextComplex64, GgswCiphertextConversionError<Self::EngineError>> {
+        Ok(unsafe { self.convert_ggsw_ciphertext_unchecked(input) })
+    }
+
+    unsafe fn convert_ggsw_ciphertext_unchecked(
+        &mut self,
+        input: &GgswCiphertext32,
+    ) -> GgswCiphertextComplex64 {
+        let input_bsk = StandardBootstrapKey::from_container(
+            input.0.as_tensor().as_container().clone(),
+            input.glwe_dimension().to_glwe_size(),
+            input.polynomial_size(),
+            input.decomposition_level_count(),
+            input.decomposition_base_log(),
+        );
+        let output = ImplFourierBootstrapKey::allocate(
+            Complex64::new(0., 0.),
+            input.glwe_dimension().to_glwe_size(),
+            input.polynomial_size(),
+            input.decomposition_level_count(),
+            input.decomposition_base_log(),
+            LweDimension(1),
+        );
+        let mut output_bsk = FourierLweBootstrapKey32(output);
+        let buffers = self.get_fourier_bootstrap_u32_buffer(&output_bsk);
+        output_bsk.0.fill_with_forward_fourier(&input_bsk, buffers);
```

Comment:
So you are in favor of copying the fill_with_forward_fourier method inside the ggsw struct? (As I did with the external product.)

Reply:
The best solution would be to also expose a `fill_with_forward_fourier` method on the ggsw struct, and to refactor the internal so that both this one and the method of the fourier bsk use the same code :slightly_smiling_face:

Reply:
I concur.